pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -59,66 +60,162 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	// FP validation
 	fpDynamic, err := dynamic.NewValidator(dv.log, dv.env, dv.env.Environment(), dv.subscriptionDoc.ID, dv.fpAuthorizer, dynamic.AuthorizerFirstParty)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to validate the first party principal",
```

Comment:
This doesn't look right. `dynamic.NewValidator` always returns `nil`. We should probably look into modifying `dynamic.NewValidator` and `dynamic.NewServicePrincipalValidator` to remove an error (I would suggest a separate PR/Work item).